lib/internal/modules/cjs/loader.js

```diff
@@ -629,9 +629,11 @@ Module.prototype.load = function(filename) {
     // Create module entry at load time to snapshot exports correctly
     const exports = this.exports;
     if (module !== undefined) { // called from cjs translator
-      module.reflect.onReady((reflect) => {
-        reflect.exports.default.set(exports);
-      });
+      if (module.reflect) {
+        module.reflect.onReady((reflect) => {
+          reflect.exports.default.set(exports);
+        });
+      }
```

Comment:
To check this properly, can we first try and isolate the bug that this PR fixes?

I suspect it may be a circular require like:

x.js
```js
require('./x.js')
```

If that definitely catches it, then we should add a test case like the following to check the above fix:

x.js
```js
module.exports = 'first';
// self-require
require('./x.js');
module.exports = 'last';
process.nextTick(() => {
  module.exports = 'too late';
});
```

Where we need to check explicitly that the value returned for the module is `last`.

Reply:
It seems that can't catch it using code above.
The value returned for the module is both `last` under v11.4.0 and v10.141.

index.js:
```js
const x = require('./x');

console.log(x);
```   

```sh
$ node -v
v11.4.0
$ node index.js  
last
$ nvm use v10                             
Now using node v10.14.1 (npm v6.4.1)
$ node index.js           
last
```

Reply:
I think I managed to isolate the big in this PR: https://github.com/nodejs/node/pull/27443